File: 24 Higher Raytracing Performance/start/model_parser/triangle.py
from config import *

class Triangle:

    def __init__(self):
        self.corners: list[np.ndarray] = []
        self.centroid: np.ndarray = pyrr.vector3.create(dtype = np.float32)

    def make_centroid(self):

        self.centroid = (self.corners[0] + self.corners[1] + self.corners[2]) / 3

File: 24 Higher Raytracing Performance/finished/model_parser/model_parser/triangle.cpp
//
//  triangle.cpp
//  model_parser
//
//  Created by Andrew Mengede on 16/2/2023.
//

#include "triangle.hpp"

void Triangle::make_centroid() {
    centroid = (corners[0] + corners[1] + corners[2]) * 1.0f / 3.0f;
}
